services/datamanager/datacapture/data_capture_queue.go

```diff
@@ -0,0 +1,81 @@
+package datacapture
+
+import (
+	"sync"
+
+	v1 "go.viam.com/api/app/datasync/v1"
+)
+
+// MaxFileSize is the maximum size in bytes of a data capture file.
+var MaxFileSize = int64(64 * 1024)
+
+// Queue is a persistent queue of SensorData backed by a series of datacapture.Files.
+type Queue struct {
+	Directory string
+	MetaData  *v1.DataCaptureMetadata
+	nextFile  *File
+	lock      *sync.Mutex
+}
+
+// NewQueue returns a new Queue.
+func NewQueue(dir string, md *v1.DataCaptureMetadata) *Queue {
+	return &Queue{
+		Directory: dir,
+		lock:      &sync.Mutex{},
+		MetaData:  md,
+	}
+}
+
+// Push pushes item onto q.
```

Comment:
Perhaps include that Push creates a binary file, or pushes SensorData into the queue & creates a file when the queue is full.

[minor] this technically doesn't need to be named a Queue anymore since it's basically a buffer that's flushed to create a file, and there's no Pop mechanism.

Reply:
Changed name from queue -> buffer, and renamed push -> write, and added to comment